roseau/load_flow/network.py

```diff
@@ -43,6 +46,96 @@
 
 
 class ElectricalNetwork:
+    """Electrical network class.
+
+    This class represents an electrical network, its elements, and their connections. After
+    creating the network, the load flow algorithm can be run on it using the
+    :meth:`solve_load_flow` method.
+
+    Args:
+        buses:
+            The buses of the network. Either a list of buses or a dictionary of buses with
+            their IDs as keys. Buses are the nodes of the network. They connect other elements
+            such as loads and sources. Buses can be connected together with branches.
+
+        branches:
+            The branches of the network. Either a list of branches or a dictionary of branches
+            with their IDs as keys. Branches are the elements that connect two buses together.
+            They can be lines, transformers, or switches.
+
+        loads:
+            The loads of the network. Either a list of loads or a dictionary of loads with their
+            IDs as keys. There are three types of loads: constant power, constant current, and
+            constant impedance.
+
+        sources:
+            The sources of the network. Either a list of sources or a dictionary of sources with
+            their IDs as keys. A network must have at least one source. Note that two sources
+            cannot be connected with a switch.
+
+        grounds:
+            The grounds of the network. Either a list of grounds or a dictionary of grounds with
+            their IDs as keys. LV networks typically have one ground element connected to the
+            neutral of the main source bus (secondary of the MV/LV transformer). HV networks
+            may have one or more grounds connected to the shunt components of their lines.
+
+        potential_refs:
+            The potential references of the network. Either a list of potential references or a
+            dictionary of potential references with their IDs as keys. As the name suggests, this
+            element defines the reference of potentials of the network. A potential reference per
+            galvanically isolated section of the network is expected. A potential reference can
+            be connected to a bus or to a ground.
+
+    Attributes:
+        DEFAULT_PRECISION (float):
+            The default precision needed for the convergence of the load flow algorithm. The solver
+            stops when the error between two iterations is below this value. Default is 1e-6.
```

Comment:
That's not "between two iterations". At each iteration, we compute the residuals of the equations of the problem. When the maximum of the absolute values of the residuals vector is lower than the provided precision, the iterations stopped.